wp-content/themes/core/components/blocks/icon_grid/Icon_Grid_Controller.php

```diff
@@ -0,0 +1,251 @@
+	private string $description;
+	private array  $cta;
+	private array  $icons;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->title             = (string) $args[ self::TITLE ];
+		$this->leadin            = (string) $args[ self::LEADIN ];
+		$this->description       = (string) $args[ self::DESCRIPTION ];
+		$this->cta               = (array) $args[ self::CTA ];
+		$this->icons             = (array) $args[ self::ICONS ];
+	}
+
+	protected function defaults(): array {
+		return [
+			self::CLASSES           => [],
+			self::ATTRS             => [],
+			self::CONTAINER_CLASSES => [],
+			self::CONTENT_CLASSES   => [],
+			self::TITLE             => '',
+			self::LEADIN            => '',
+			self::DESCRIPTION       => '',
+			self::CTA               => [],
+			self::ICONS             => [],
+		];
+	}
+
+	protected function required(): array {
+		return [
+			self::CLASSES           => [ 'c-block', 'b-icon-grid' ],
+			self::CONTAINER_CLASSES => [ 'l-container' ],
+		];
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_classes(): string {
+		return Markup_Utils::class_attribute( $this->classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_attrs(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		$this->content_classes[] = 'g-3-up';
+		$this->content_classes[] = 'g-centered';
+
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
+			return [];
+		}
+
+		return [
+			Content_Block_Controller::TAG     => 'header',
+			Content_Block_Controller::LAYOUT  => Content_Block_Controller::LAYOUT_CENTER,
+			Content_Block_Controller::LEADIN  => $this->get_leadin(),
+			Content_Block_Controller::TITLE   => $this->get_title(),
+			Content_Block_Controller::CONTENT => $this->get_content(),
+			Content_Block_Controller::CTA     => $this->get_cta(),
+			Content_Block_Controller::CLASSES => [
+				'c-block__content-block',
+				'c-block__header',
+				'b-icon-grid__header',
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_leadin(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'c-block__leadin',
+				'b-icon-grid__leadin',
+				'h6',
+			],
+			Text_Controller::CONTENT => $this->leadin ?? '',
```

Comment:
```suggestion
			Text_Controller::CONTENT => $this->leadin,
```
You're setting defaults in the constructor, so the `??` operator isn't needed